src/measurements/CalibrationClasses.py

```
@@ -342,25 +342,24 @@ def __init__(self,devices, background, grating_period, beam_, compression_carrie
             'wavelengths' : self.wls,
             'intensities' : self.intensities
         }
-        # Configure single beam over which the columns will be scanned
-
-        self.monobeam=Beam(self.SLM.get_width(),self.SLM.get_height())
 
-        self.monobeam.set_pixelToWavelength(Polynomial(1e-9*np.array([compression_carrier_wavelength-100,1/10]))) 
-        self.monobeam.set_compressionCarrierWave(compression_carrier_wavelength*1e-9) 
-        self.monobeam.set_gratingPeriod(grating_period)
-
-        self.chirp_ = np.linspace(chirp_max,chirp_min,num=int((chirp_max-chirp_min)/chirp_step))
+        self.chirp = np.linspace(chirp_max,chirp_min,num=int((chirp_max-chirp_min)/chirp_step))
         self.isDemo= demo
-        self.beam_ = beam
-        self.beam_.set_pixelToWavelength(Polynomial(1e-9*np.array([compression_carrier_wavelength-100,1/10])))
-        self.beam_.set_compressionCarrierWave(compression_carrier_wavelength*1e-9) 
-        self.beam_.set_gratingPeriod(grating_period)
+        self.beam_name = beam_name
+        self.beam = beam
+        if spectral_calibration == None:
+            self.beam.set_pixelToWavelength(Polynomial(1e-9*np.array([compression_carrier_wavelength-100,1/10]))) # arbitrairy polynomial spectral calibration
```

What is the logic here? If no spectral calibration is present, why not simply raise an exception and abort the temporal calibration?

---

That's right. The last topic get your point. However, it will used an arbitrary spectral calibration instead of abording the temporal calibration.

---

Why perform a temporal calibration if there is no spectral calibration? I insist, if there is no point, it should raise an error.